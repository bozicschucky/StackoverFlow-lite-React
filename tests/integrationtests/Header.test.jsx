import { shallow } from "enzyme";
import React from "react";
import Header from '../../src/components/header';

describe("Tests the mounted Header component", () => {
  it("tests the Header component against the snapshot", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
