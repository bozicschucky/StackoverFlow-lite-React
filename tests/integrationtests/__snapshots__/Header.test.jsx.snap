// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the mounted Header component tests the Header component against the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav-bar"
      >
        <div>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            MiniOverFlow
          </NavLink>
        </div>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            MiniOverFlow
          </NavLink>
        </div>,
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            MiniOverFlow
          </NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "MiniOverFlow",
            "to": "/",
          },
          "ref": null,
          "rendered": "MiniOverFlow",
          "type": [Function],
        },
        "type": "div",
      },
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav-bar"
        >
          <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              MiniOverFlow
            </NavLink>
          </div>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              MiniOverFlow
            </NavLink>
          </div>,
          "className": "nav-bar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
            >
              MiniOverFlow
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "MiniOverFlow",
              "to": "/",
            },
            "ref": null,
            "rendered": "MiniOverFlow",
            "type": [Function],
          },
          "type": "div",
        },
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
